File: AddonTemplateVb/addonClass.vb
Imports System
Imports System.Collections.Generic
Imports System.Text
Imports Contensive.BaseClasses

Namespace Contensive.Addons
    '
    ' Sample Vb addon
    '
    Public Class HelloWorldClass
        Inherits AddonBaseClass
        '
        ' - update references to your installed version of cpBase
        ' - Edit project - under application, verify root name space is empty
        ' - Change the namespace in this file to the collection name
        ' - Change this class name to the addon name
        ' - Create a Contensive Addon record, set the dotnet class full name to yourNameSpaceName.yourClassName
        '
        '=====================================================================================
        ' addon api
        '=====================================================================================
        '
        Public Overrides Function Execute(ByVal CP As CPBaseClass) As Object
            Dim returnHtml As String
            Try
                returnHtml = "Visual Studio Contensive Addon - OK response"
            Catch ex As Exception
                errorReport(CP, ex, "execute")
                returnHtml = "Visual Studio Contensive Addon - Error response"
            End Try
            Return returnHtml
        End Function
        '
        '=====================================================================================
        ' common report for this class
        '=====================================================================================
        '
        Private Sub errorReport(ByVal cp As CPBaseClass, ByVal ex As Exception, ByVal method As String)
            Try
                cp.Site.ErrorReport(ex, "Unexpected error in sampleClass." & method)
            Catch exLost As Exception
                '
                ' stop anything thrown from cp errorReport
                '
            End Try
        End Sub
    End Class
End Namespace

File: AddonTemplateCs/addonClass.cs
using System;
using System.Collections.Generic;
using System.Text;
using Contensive.BaseClasses;

namespace addonTemplateCs
{
    //
    // 1) Change the namespace to the collection name
    // 2) Change this class name to the addon name
    // 3) Create a Contensive Addon record with the namespace apCollectionName.ad
    // 3) add reference to CPBase.DLL, typically installed in c:\program files\kma\contensive\
    //
    public class addonClass : Contensive.BaseClasses.AddonBaseClass
    {
        //
        // execute method is the only public
        //
        public override object Execute(Contensive.BaseClasses.CPBaseClass cp)
        {
            string returnHtml = "";
            try
            {
                returnHtml = "Visual Studio add-on assembly - OK response";
            }
            catch (Exception ex)
            {
                cp.Site.ErrorReport( "There was a unexpected exception, " + ex.ToString() );
                returnHtml = "Visual Studio add-on assembly - error response";
            }
            return returnHtml;
        }
    }
}
